app/code/Tsg/Improvements/Ui/Component/DataProvider/Grid.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Tsg\Improvements\Ui\Component\DataProvider;
+
+use Magento\Framework\View\Element\UiComponent\DataProvider\DataProvider;
+use Magento\Framework\Api\Search\ReportingInterface;
+use Magento\Framework\Api\Search\SearchCriteriaBuilder;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Api\FilterBuilder;
+use Magento\Framework\UrlInterface;
+use Tsg\Improvements\Configs;
+
+class Grid extends DataProvider
+{
+    private $fileScanner;
+
+    public function __construct(
+        $name,
+        ReportingInterface $reporting,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        RequestInterface $request,
+        FilterBuilder $filterBuilder,
+        UrlInterface $url,
+        FileScanner $fileScanner
+
+    ) {
+        $primaryFieldName = 'id';
+        $requestFieldName = 'id';
+        $meta = [];
+        $updateUrl = $url->getUrl('mui/index/render');
+        $data = [
+            'config' => [
+                'component' => 'Magento_Ui/js/grid/provider',
+                'update_url' => $updateUrl
+            ]
+        ];
+
+        parent::__construct($name, $primaryFieldName, $requestFieldName, $reporting, $searchCriteriaBuilder, $request,
```

Comment:
```suggestion
        parent::__construct($name, $primaryFieldName, $requestFieldName, $reporting, $searchCriteriaBuilder, $request,
```
        parent::__construct(
            $name, 
            $primaryFieldName, 
            $requestFieldName, 
            $reporting, 
            $searchCriteriaBuilder, 
            $request,
)